### PR TITLE
Download C++ API through CMake and remove python script

### PR DESCRIPTION
**Pull request by rLinks234**

### CODE REVIEW

File: projects/cmake/DownloadHebiCpp.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Used to download the C++ API - this should not be used directly.
+cmake_minimum_required(VERSION 3.0)
+
+set(HEBI_CPP_VERSION "2.0.1")
+set(HEBI_CPP_FILE_NAME "hebi-cpp-${HEBI_CPP_VERSION}.tar.gz")
+set(HEBI_CPP_LIB_SHA256 "9930dc65b543510bd211a98ba493482ddb9841afa57ae1f3fb958c6d35a975c7")
```
Comment (rLinks234) [inline projects/cmake/DownloadHebiCpp.cmake]: We'll need to update this per C++ version bump, but it's good for us to do

File: projects/cmake/DownloadHebiCpp.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+set(HEBI_CPP_URL "http://docs.hebi.us/download/cpp/${HEBI_CPP_FILE_NAME}")
+
+# If the CMakeLists.txt is not found, then redownload the C++ API 
+if(NOT EXISTS ${HEBI_DIR}/CMakeLists.txt)
+
+message(STATUS "Downloading C++ API version ${HEBI_CPP_VERSION} from ${HEBI_CPP_URL}")
+set(HEBI_CPP_FILE_FULL_PATH "${CMAKE_BINARY_DIR}/${HEBI_CPP_FILE_NAME}")
+file(DOWNLOAD "${HEBI_CPP_URL}" "${HEBI_CPP_FILE_FULL_PATH}"
```
Comment (rLinks234) [inline projects/cmake/DownloadHebiCpp.cmake]: This exists all the way back to [version 3.0](https://cmake.org/cmake/help/v3.0/command/file.html) of cmake.

File: projects/cmake/DownloadHebiCpp.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  EXPECTED_HASH SHA256=${HEBI_CPP_LIB_SHA256}
+  SHOW_PROGRESS)
+
+# Try and find a python install
+find_program(PYTHON_EXECUTABLE python)
+
+if(EXISTS "${PYTHON_EXECUTABLE}")
+  execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c"
```
Comment (rLinks234) [inline projects/cmake/DownloadHebiCpp.cmake]: Unfortunately, this is the easiest way I can find to do this for Windows. `tarfile` is a package which comes with a python installation - you don't need to install it through `pip`, etc

File: projects/cmake/DownloadHebiCpp.cmake [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "import tarfile; tar = tarfile.open(r'${HEBI_CPP_FILE_FULL_PATH}'); tar.extractall(path=r'${ROOT_DIR}'); tar.close()")
+# Fall back on `tar` for MacOS and Linux
+elseif(NOT EXISTS "${PYTHON_EXECUTABLE}" AND NOT WIN32)
+  execute_process(COMMAND "tar" "-xvf" "${HEBI_CPP_FILE_FULL_PATH}"
+    WORKING_DIRECTORY ${ROOT_DIR})
+elseif(WIN32)
+  message(FATAL_ERROR "Cannot extract ${HEBI_CPP_FILE_FULL_PATH} because python was not found in the executable path. Extract this file manually and re-run CMake.")
```
Comment (rLinks234) [inline projects/cmake/DownloadHebiCpp.cmake]: I'd love to change this someday, but as we've seen, Windows does not have any easy way to extract zips/tarballs from the command line..

File: projects/cmake/DownloadHebiCpp.cmake (hunk 1 above)
Comment (iamtesch) [reply projects/cmake/DownloadHebiCpp.cmake]: :+1:

**Pull request merged**